SpringProject/src/it/java/com/gradproject2019/user/UserControllerIT.java

```diff
@@ -139,15 +143,59 @@ public void shouldReturn200WhenGettingUserById() {
     }
 
     @Test
-    public void shouldReturn401WhenGettingUserByIDUnauthorised() throws URISyntaxException {
-        URI uri = new URI(baseUri + "/" + savedUser.getId());
+    public void shouldReturn200WhenGettingManagerById() {
+        User employee = userRepository.saveAndFlush(new User("employee", "employee", "employee", "employee@email.com", "Password!1", "employee", savedUser.getId()));
+        Token employeeToken = authRepository.saveAndFlush(new Token(employee.getId(), UUID.randomUUID()));
+        HttpHeaders employeeHeaders = new HttpHeaders();
+        employeeHeaders.add(HttpHeaders.AUTHORIZATION, employeeToken.getToken().toString());
 
+        ResponseEntity<User> response = getUserById(uri2, employeeHeaders);
+
+        Assert.assertEquals(200, response.getStatusCodeValue());
+        Assert.assertEquals(savedUser.getId(), response.getBody().getId());
+        Assert.assertEquals(savedUser.getFirstName(), response.getBody().getFirstName());
+        Assert.assertEquals(savedUser.getUsername(), response.getBody().getUsername());
+        Assert.assertEquals(savedUser.getLastName(), response.getBody().getLastName());
+        Assert.assertEquals(savedUser.getEmail(), response.getBody().getEmail());
+        Assert.assertEquals(savedUser.getOccupation(), response.getBody().getOccupation());
+    }
+
+    @Test
+    public void shouldReturn200WhenGettingEmployeeById() throws URISyntaxException {
+        User employee = userRepository.saveAndFlush(new User("employee", "employee", "employee", "employee@email.com", "Password!1", "employee", savedUser.getId()));
+        URI employeeUri = new URI(baseUri + "/" + employee.getId());
+
+        ResponseEntity<User> response = getUserById(employeeUri, passingHeaders);
+
+        Assert.assertEquals(200, response.getStatusCodeValue());
+        Assert.assertEquals(employee.getId(), response.getBody().getId());
+        Assert.assertEquals(employee.getFirstName(), response.getBody().getFirstName());
+        Assert.assertEquals(employee.getUsername(), response.getBody().getUsername());
+        Assert.assertEquals(employee.getLastName(), response.getBody().getLastName());
+        Assert.assertEquals(employee.getEmail(), response.getBody().getEmail());
+        Assert.assertEquals(employee.getOccupation(), response.getBody().getOccupation());
+    }
+
+    @Test
+    public void shouldReturn401WhenGettingUserByIdUnauthorisedDueToToken() {
         ResponseEntity<ErrorEntity> response = getUserByIdExpectingAuthError();
 
         Assert.assertEquals(401, response.getStatusCodeValue());
         Assert.assertEquals("User unauthorized to perform action.", response.getBody().getMessage());
     }
 
+    @Test
+    public void shouldReturn403WhenGettingUserWhoIsNotTheUserOrTheirManagerOrTheirEmployee() throws URISyntaxException {
```

Comment:
This sounds really confusing, how about just shouldReturn403WhenGettingUserForbidden? Imagine you change that logic later, you'll have to also change the name of this test.

Reply:
True, I didn't really like the name when I wrote it, this does read better.